src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -606,6 +711,128 @@ static void Test2(object y, out int x)
         }
 
         [Fact]
+        public void Scope_02()
+        {
+            var text = @"
+public class Cls
+{
+    public static void Main()
+    {
+        Test2(x1, 
+              Test1(out int x1));
+    }
+
+    static object Test1(out int x)
+    {
+        x = 1;
+        return null;
+    }
+
+    static void Test2(object y, object x)
+    {
+    }
+}";
+            var compilation = CreateCompilationWithMscorlib(text, options: TestOptions.ReleaseExe, parseOptions: TestOptions.Regular.WithOutVarFeature());
+
+            compilation.VerifyDiagnostics(
+                // (6,15): error CS0841: Cannot use local variable 'x1' before it is declared
```

Comment:
I'm suggesting you deleted the only constraint on where the variable may be used, and then you implemented a constraint. The result is that the spec and implementation disagree (i.e. a bug). This can be fixed in one of four ways: fixing the spec, adding an issue in .work.md, opening an issue, or removing the constraint.

---
In reply to: [64252591](https://github.com/dotnet/roslyn/pull/11493#discussion_r64252591) [](ancestors = 64252591,64250908)